doc/dev.md

```diff
@@ -0,0 +1,30 @@
+# Developer documentation
+
+## GHC versions
+
+We support the three most recent versions of GHC.
+We try to support new versions as soon as they are supported by the libraries that we depend on.
+
+### Adding a new version
+
+The following checklist enumerates the steps needed to support a new version of GHC.
+When performing such an upgrade, it can be helpful to copy/paste this list into the MR description and check off what has been done, so as to not forget anything.
+
+- [ ] Allow the [new version of `base`][base] in the Cabal `build-depends`
+- [ ] Run `cabal {build,test,haddock}`, bumping dependency bounds as needed
```

Comment:
I think this checklist needs to say "update the submodules to include any necessary changes to support newer GHC versions" (or something to that effect). That ends up taking quite a bit of time when adding support for newer GHC versions.

Reply:
Just so I understand are you saying that we should mention that some of our deps are submodules and so need to be bumped using `git` rather than in the Cabal file? Or are you saying that some of our deps are also Galois-owned and will need to support the new GHC version before Crucible can? Or both? (Feel free to suggest wording, or just answer the above and I can give it a go.)

Reply:
Both, IMO.